### PR TITLE
gui win acquisition OverviewAcquisitionDialog: improve area computation reliability

### PR DESCRIPTION
**Pull request by pieleric**
If the stage was at the wrong place, an error would be reported... and
then the rest would fail with an exception.
=> Now handle properly when area cannot be computed by setting .area to None

Also handle zlevels computation corner cases when it's near the focus
range.

Also rearange a little bit the function organization to make it clearer.

Also update the test cases to not depend on a simulator, and actually
pass. (got broken when the simulator was adjusted)

Also remove some unneeded code.

### CODE REVIEW

File: src/odemis/gui/win/acquisition.py [hunk 1]
```diff
@@ -891,9 +885,33 @@ def update_acquisition_time(self, acq_time=999):
...
         self.lbl_acqestimate.SetLabel(txt)
 
+    def get_fov(self, s):
+        try:
+            return s.guessFoV()
+        except (NotImplementedError, AttributeError):
+            raise TypeError("Unsupported Stream %s, it doesn't have a .guessFoV()" % (s,))
+
+    @staticmethod
+    def get_ROA_rect(w, h, pos, tiling_rng):
```
[the remark this answers is not in the document]
Comment (pieleric) [reply src/odemis/gui/win/acquisition.py]: I couldn't find a good name. That's indeed much better than the current name.

**Review by tepals — APPROVED** [review]
